### ISSUE TITLE
tidy version-specific IE legacy quirks

### ISSUE THREAD

**Issue by jonassmedegaard** [Compass/compass#868]
I noticed this recent commit to compass-susy-plugin moving away from a compass-provided snippet and instead reimplementing:
https://github.com/ericam/susy/commit/b7598c

These seem to be affected (based on a guick'n'dirty grep for IE in capital letters):

stylesheets/compass/css3/_images.scss
stylesheets/compass/css3/_transform.scss
stylesheets/compass/typography/_vertical_rhythm.scss
stylesheets/compass/typography/lists/_horizontal-list.scss
stylesheets/compass/typography/lists/_inline-list.scss
stylesheets/compass/utilities/general/_float.scss
stylesheets/compass/utilities/general/_hacks.scss
stylesheets/compass/utilities/tables/_borders.scss

**Comment by jonassmedegaard** [comment]
Reading again I guess I am not at all clear what I sugest here, so let me try clarify:

I suggest to apply IE quirks only when $legacy-support-for-ie is true, or for quirks specific to particular version(s) of IE only when the correspondening variable is true.

**Comment by mirisuzanne** [comment]
I agree, although in this case the only reason to use the mixin is for legacy IE6 support, so I wasn't sure if the mixin needed to change. If you don't need to support IE6, you can simply use `float` without the mixin. I suppose either way it shoul be documented.

**Comment by chriseppstein** [comment]
That commit link is broken so I can't sort out what this conversation is quite about. @ericam is there any action needed here?

**Comment by mirisuzanne** [comment]
I updated the link. Yes, there are a few places this change could be implemented.

In brief, we have a few legacy hacks that are not checking the legacy-support variables. I'll slog through and see if I can find/fix them all. Shouldn't take long.

**Comment by chriseppstein** [comment]
@ericam you rock.

**Comment by mirisuzanne** [comment]
@chriseppstein: This is odd. When I check `$legacy-support-for-ie6` in _float.scss, I get this error when running tests:

``` bash
Error: test_on_stylesheet_saved_callback(CompassTest)
Sass::SyntaxError: Undefined variable: "$legacy-support-for-ie6".
/Users/eric/Sites/compass/frameworks/compass/stylesheets/compass/utilities/general/_float.scss:15:in `float'
     12: // [double-margin bug in IE5/6](http://www.positioniseverything.net/explorer/doubled-margin.html)
     13: @mixin float($side: left) {
     14:   float: unquote($side);
  => 15:   @if $legacy-support-for-ie6 {
     16:     display: inline;} }
     17: 
     18: // Resets floated elements back to their default of `float: none` and defaults
```

no other files that reference ie6 support are throwing that error. Ideas?

**Comment by chriseppstein** [comment]
Does it import the browser support module?

**Comment by mirisuzanne** [comment]
What? You think I'm that stoopid? 

(I am. Fixed. Thanks.)

Closed by 1c9883360c

**Comment by chriseppstein** [comment]
We all have our "forest for the trees" moments :)
